pkg/scheduler/framework/types.go

```diff
@@ -238,6 +240,12 @@ func (f *FitError) Error() string {
 		}
 	}
 
+	reasonMsg := fmt.Sprintf(NoNodeAvailableMsg, f.NumAllNodes)
+	// Add the messages from PreFilter to reasonMsg.
+	preFilterMsg := f.Diagnosis.PreFilterMsg
+	if preFilterMsg != "" {
+		reasonMsg += preFilterMsg + " "
+	}
```

Comment:
seem we miss `:` here? as I suppose the eventual message would be like:

```
0/X nodes are available: <PreFilterMsg> <FilterMsg> <PostFilterMsg>
```

Given PreFilterMsg and PostFilterMsg can be empty, so we can start the message as `0/X nodes are available:`, and then append "the empty space plus <Msg>" as a chunk afterwards.

Reply:
PS1: can we add tests to verify the eventual msg format?
PS2: empty space doesn't look like an appropriate separtor, maybe we should use something like `|` to separate PreFitler/Filter/PostFilter message.

Reply:
Updated the format. How about using `.` as the separator. 

`0/X nodes are available: <PreFilterMsg>. <FilterMsg>. <PostFilterMsg>.`